validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -86,32 +91,43 @@ def _receive_message(self):
             message = validator_pb2.Message()
             message.ParseFromString(msg_bytes)
 
-            LOGGER.debug("receiving %s message",
-                         get_enum_name(message.message_type))
+            LOGGER.debug("%s receiving %s message: %s bytes",
+                         self._connection,
+                         get_enum_name(message.message_type),
+                         sys.getsizeof(msg_bytes))
+
             try:
                 self._futures.set_result(
                     message.correlation_id,
                     future.FutureResult(message_type=message.message_type,
                                         content=message.content))
             except future.FutureCollectionKeyError:
                 if self._socket.getsockopt(zmq.TYPE) == zmq.ROUTER:
-                    self._dispatcher.dispatch(identity, message)
+                    self._dispatcher.dispatch(self._connection,
+                                              identity,
+                                              message)
                 else:
-                    LOGGER.info(
-                        "received a first message on the zmq dealer.")
+                    # Because this is a zmq.DEALER socket, there is no
+                    # outbound identity
+                    self._dispatcher.dispatch(self._connection,
+                                              None,
```

Comment:
If identity is something optional, should the dispatch method's arguments be arranged as such (with any defaults for identity as needed).  This would match our other uses of Identity in the system.